fdbrpc/FlowTransport.actor.cpp

```diff
@@ -696,19 +692,35 @@ static void scanPackets(TransportData* transport, uint8_t*& unprocessed_begin, u
 	}
 }
 
+// Given unprocessed buffer [begin, end), check if next packet size is known and return
+// enough size for the next packet, whose format is: {size, optional_checksum, data}.
+static int getNewBufferSize(const uint8_t* begin, const uint8_t* end, const NetworkAddress& peerAddress) {
```

Comment:
For workloads with packets less than 4k, we will start with a buffer of 64k which will leave a partial result, and then after that we will use 4k buffers, which will also have partial results so we will continue with 4k buffers. 

For workloads with packets larger than 4k, we will cycle between a buffer of 64k, followed by an exact size buffer, followed by another buffer of 64k etc. 

It seems like the correct behavior if we have a len less than sizeof(uint32_t) we should enough such that len==sizeof(uint32_t)